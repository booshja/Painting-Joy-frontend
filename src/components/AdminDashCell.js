// dependencies
import React from "react";
import styled from "styled-components";
// components
import { StyledCell } from "../containers/styles/adminContainers";
import { StyledP } from "../containers/styles/adminTypography";
import { Link } from "react-router-dom";

const StyledAdminDashCell = styled.div`
    display: flex;
    flex-direction: column;
    margin-bottom: 1rem;
    width: 95%;
    align-self: flex-end;
`;

const StyledDashCell = styled(StyledCell)`
    width: 95%;
    border-width: 2px;
    background-color: ${(props) => props.bgColor};
`;

const StyledBoldP = styled(StyledP)`
    font-weight: 700;
    font-size: 1.4rem;
    color: #ffffff;
`;

const StyledLink = styled(Link)`
    font-family: "News Cycle", sans-serif;
    font-size: 1.4rem;
    color: #ffffff;
    font-weight: 700;
    letter-spacing: 1px;
`;

const AdminDashCell = ({ children, title, linkTo, footer, data }) => {
    return (
        <StyledAdminDashCell>
            <StyledDashCell bgColor="#db9487">
                <StyledBoldP>{title}</StyledBoldP>
            </StyledDashCell>
            <StyledDashCell bgColor="#ffffff">{children}</StyledDashCell>
            {footer ? (
                <StyledDashCell bgColor="#db9487">
                    <StyledLink to={linkTo}>See All &gt;</StyledLink>
                </StyledDashCell>
            ) : null}
        </StyledAdminDashCell>
    );
};

export default AdminDashCell;
