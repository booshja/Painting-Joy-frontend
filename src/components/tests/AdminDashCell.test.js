import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import AdminDashCell from "../AdminDashCell";
import "jest-styled-components";

/** Smoke Test */
it("renders without crashing, footer false", () => {
    render(
        <BrowserRouter>
            <AdminDashCell
                children={<h1>testy</h1>}
                title="This is a title"
                linkTo="/linkhere"
                footer={false}
            />
        </BrowserRouter>
    );
});

it("renders without crashing, footer true", () => {
    render(
        <BrowserRouter>
            <AdminDashCell
                children={<h1>testy</h1>}
                title="This is a title"
                linkTo="/linkhere"
                footer={true}
            />
        </BrowserRouter>
    );
});

/** Snapshot Test */
it("matches snapshot, footer true", () => {
    const { asFragment } = render(
        <BrowserRouter>
            <AdminDashCell
                children={<h1>testy</h1>}
                title="This is a title"
                linkTo="/linkhere"
                footer={true}
            />
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot, footer false", () => {
    const { asFragment } = render(
        <BrowserRouter>
            <AdminDashCell
                children={<h1>testy</h1>}
                title="This is a title"
                linkTo="/linkhere"
                footer={false}
            />
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Tests */
it("renders correctly, footer true", () => {
    const { getByText } = render(
        <BrowserRouter>
            <AdminDashCell
                children={<h1>testy</h1>}
                title="This is a title"
                linkTo="/linkhere"
                footer={true}
            />
        </BrowserRouter>
    );
    const title = getByText("This is a title");
    const children = getByText("testy");
    const link = getByText("See All >");

    expect(title).toBeInTheDocument();
    expect(children).toBeInTheDocument();
    expect(link).toBeInTheDocument();
});

it("renders correctly, footer false", () => {
    const { getByText } = render(
        <BrowserRouter>
            <AdminDashCell
                children={<h1>testy</h1>}
                title="This is a title"
                linkTo="/linkhere"
                footer={false}
            />
        </BrowserRouter>
    );
    const title = getByText("This is a title");
    const children = getByText("testy");

    expect(title).toBeInTheDocument();
    expect(children).toBeInTheDocument();
});
