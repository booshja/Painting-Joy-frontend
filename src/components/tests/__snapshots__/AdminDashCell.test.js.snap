// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot, footer false 1`] = `
<DocumentFragment>
  .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  border: 1px solid #207070;
  width: 100%;
  padding: 1rem;
  background-color: #ffffff;
}

.c3 {
  font-size: 1.2rem;
  font-family: "News Cycle",sans-serif;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 1rem;
  width: 95%;
  max-width: 600px;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
}

.c2 {
  width: 95%;
  border-width: 2px;
  background-color: #db9487;
}

.c5 {
  width: 95%;
  border-width: 2px;
  background-color: #ffffff;
}

.c4 {
  font-weight: 700;
  font-size: 1.4rem;
  color: #ffffff;
}

@media screen and (min-width:1024px) {
  .c1 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
  }
}

@media screen and (min-width:1024px) {
  .c0 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
  }
}

@media screen and (min-width:1024px) {
  .c0 {
    -webkit-align-self: flex-start;
    -ms-flex-item-align: start;
    align-self: flex-start;
  }
}

@media screen and (min-width:1024px) {
  .c0 {
    width: 50%;
  }
}

<div
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <p
        class="c3 c4"
      >
        This is a title
      </p>
    </div>
    <div
      class="c1 c5"
    >
      <h1>
        testy
      </h1>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot, footer true 1`] = `
<DocumentFragment>
  .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  border: 1px solid #207070;
  width: 100%;
  padding: 1rem;
  background-color: #ffffff;
}

.c3 {
  font-size: 1.2rem;
  font-family: "News Cycle",sans-serif;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 1rem;
  width: 95%;
  max-width: 600px;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
}

.c2 {
  width: 95%;
  border-width: 2px;
  background-color: #db9487;
}

.c5 {
  width: 95%;
  border-width: 2px;
  background-color: #ffffff;
}

.c4 {
  font-weight: 700;
  font-size: 1.4rem;
  color: #ffffff;
}

.c6 {
  font-family: "News Cycle",sans-serif;
  font-size: 1.4rem;
  color: #ffffff;
  font-weight: 700;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c6:hover {
  font-style: italic;
}

@media screen and (min-width:1024px) {
  .c1 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
  }
}

@media screen and (min-width:1024px) {
  .c0 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
  }
}

@media screen and (min-width:1024px) {
  .c0 {
    -webkit-align-self: flex-start;
    -ms-flex-item-align: start;
    align-self: flex-start;
  }
}

@media screen and (min-width:1024px) {
  .c0 {
    width: 50%;
  }
}

<div
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <p
        class="c3 c4"
      >
        This is a title
      </p>
    </div>
    <div
      class="c1 c5"
    >
      <h1>
        testy
      </h1>
    </div>
    <div
      class="c1 c2"
    >
      <a
        class="c6"
        href="/linkhere"
      >
        See All &gt;
      </a>
    </div>
  </div>
</DocumentFragment>
`;
